File: ros_basics/src/float_subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/Float32.h"
#include <iostream>

void callback(const std_msgs::Float32::ConstPtr& msg)
{
	ROS_INFO("Subscriber received: %f", msg->data);
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "float_subscriber");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/float",10,callback);

	ros::spin();

	return 0;
}

File: ros_basics/scripts/float_subscriber.py
#!/usr/bin/env python

import rospy 
from std_msgs.msg import Float32

def callback(data):
	rospy.loginfo("Subscriber heard : %f", data.data)

def subscriber():
	rospy.init_node('float_subscriber', anonymous=True)
	rospy.Subscriber("float", Float32, callback)

	rospy.spin()

if __name__ == '__main__':
	try:
		subscriber()
	except rospy.ROSInterruptException():
		pass